demos/projects/ESPRESSIF/adu/port/azure_iot_flash_platform.c

```diff
@@ -71,6 +71,12 @@ AzureIoTResult_t AzureIoTPlatform_Init( AzureADUImage_t * const pxAduImage )
     return eAzureIoTSuccess;
 }
 
+int64_t AzureIoTPlatform_GetSingleFlashBootBankSize()
+{
+    /* size of the next ota partition to be written to */
+    return esp_ota_get_next_update_partition( esp_ota_get_running_partition() )->size;
```

Comment:
can this return `NULL`? If so might have a hard fault here.

Reply:
Good catch! I updated it to check, and return -1 as the size if we run into issues. This will then fail the check that the bank size is larger than the update and reject it. It won't automatically try to get the update again, but the user can retry from the azure portal (or if the device restarts). Getting the device to fail the way we do in other steps (which would restart the device and try again) isn't super feasible at the accept/reject step without significant refactoring, so this seemed like a good compromise for a solution.